### PR TITLE
Randomly generate telnet credentials by default

### PR DESCRIPTION
**Pull request by hcoura**

### CODE REVIEW

File: scrapy/extensions/telnet.py [hunk 1]
```diff
@@ -50,8 +52,15 @@ def __init__(self, crawler):
         self.noisy = False
         self.portrange = [int(x) for x in crawler.settings.getlist('TELNETCONSOLE_PORT')]
         self.host = crawler.settings['TELNETCONSOLE_HOST']
+
         self.username = crawler.settings['TELNETCONSOLE_USERNAME']
         self.password = crawler.settings['TELNETCONSOLE_PASSWORD']
+
+        if not self.password:
+            self.password = binascii.hexlify(os.urandom(8)).decode('utf8')
+            logger.info('Telnet Username: %s', self.username)
```
Comment (dangra) [inline scrapy/extensions/telnet.py]: Do not log username

File: scrapy/extensions/telnet.py [hunk 2]
```diff
@@ -50,8 +52,14 @@ def __init__(self, crawler):
         self.noisy = False
         self.portrange = [int(x) for x in crawler.settings.getlist('TELNETCONSOLE_PORT')]
         self.host = crawler.settings['TELNETCONSOLE_HOST']
+
```
Comment (dangra) [inline scrapy/extensions/telnet.py]: picky sorry, but any reason for the extra line?

File: tests/test_extension_telnet.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    def _get_console_and_portal(self, settings=None):
+        crawler = get_crawler(settings_dict=settings)
+        console = TelnetConsole(crawler)
+        username = console.username
+        password = console.password
+
+        def _get_telnet_vars():
+            # This function has some side effects we don't need for this test
+            return {}
+        console._get_telnet_vars = _get_telnet_vars
```
Comment (dangra) [inline tests/test_extension_telnet.py]: Doesn't lambda work?

`console._get_telnet_vars = lambda: {}`

File: scrapy/extensions/telnet.py (hunk 2 above)
Comment (hcoura) [reply scrapy/extensions/telnet.py]: None, likely my brain separating things

**Pull request merged**